$OpenBSD: patch-src_hotspot_share_oops_metadata_hpp,v 1.1 2022/01/08 15:05:30 kurt Exp $

Work around using 'this' as a tagged pointer by preventing optimization
with clang 13.

Index: src/hotspot/share/oops/metadata.hpp
--- src/hotspot/share/oops/metadata.hpp.orig
+++ src/hotspot/share/oops/metadata.hpp
@@ -37,6 +37,9 @@ class Metadata : public MetaspaceObj {
   NOT_PRODUCT(Metadata() : _valid(0) {})
   NOT_PRODUCT(bool is_valid() const { return _valid == 0; })
 
+#if defined(__clang_major__) && (__clang_major__ >= 13)
+  NOINLINE
+#endif
   int identity_hash()                { return (int)(uintptr_t)this; }
 
   virtual bool is_metadata()           const { return true; }
